x/mongo/driver/auth/creds/awscreds.go

```diff
@@ -8,181 +8,48 @@ package creds
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
-	"io/ioutil"
 	"net/http"
-	"os"
 	"time"
 
-	"go.mongodb.org/mongo-driver/x/mongo/driver/auth/internal/awsv4"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+	awscredproviders "go.mongodb.org/mongo-driver/x/mongo/driver/auth/creds/aws_credential_providers"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/auth/internal/aws/credentials"
 )
 
-type ecsResponse struct {
-	AccessKeyID     string `json:"AccessKeyId"`
-	SecretAccessKey string `json:"SecretAccessKey"`
-	Token           string `json:"Token"`
-}
-
 const (
-	awsRelativeURI     = "http://169.254.170.2/"
-	awsEC2URI          = "http://169.254.169.254/"
-	awsEC2RolePath     = "latest/meta-data/iam/security-credentials/"
-	awsEC2TokenPath    = "latest/api/token"
-	defaultHTTPTimeout = 10 * time.Second
+	expiryWindow = 5 * time.Minute
```

Comment:
My understanding is that `expiryWindow` is a "grace period" that gives the driver time to refresh the credentials before they actually expire. Is that correct?

If so, what is the typical valid duration of credentials for different AWS services? Is it possible that the valid duration is ever below 5 minutes, which would effectively cause the Go driver to re-fetch the credentials every time they're requested?

Reply:
Thanks for the explanation!